codec-http/src/main/java/io/netty/handler/codec/http/cookie/DefaultCookie.java

```diff
@@ -153,10 +153,8 @@ public boolean equals(Object o) {
             if (that.domain() != null) {
                 return false;
             }
-        } else if (that.domain() == null) {
-            return false;
         } else {
-            return domain().equalsIgnoreCase(that.domain());
+            return that.domain() != null && domain().equalsIgnoreCase(that.domain());
```

Comment:
you can remove `that.domain() != null &&` as this will also be checked by `equalsIgnoreCase`